File: 154.find-minimum-in-rotated-sorted-array-ii.cpp
/*
 * @lc app=leetcode id=154 lang=cpp
 *
 * [154] Find Minimum in Rotated Sorted Array II
 */

// @lc code=start
auto speedup = [](){
  cin.tie(nullptr);
  cout.tie(nullptr);
  ios::sync_with_stdio(false);
  return 0;
}();
class Solution {
public:
  int findMin(vector<int>& nums) {
    return *min_element(nums.begin(), nums.end());
   }
};

// Accepted
// 192/192 cases passed (11 ms)
// Your runtime beats 22.47 % of cpp submissions
// Your memory usage beats 11.51 % of cpp submissions (12.3 MB)
// @lc code=end

File: 154.find-minimum-in-rotated-sorted-array-ii.js
/*
 * @lc app=leetcode id=154 lang=javascript
 *
 * [154] Find Minimum in Rotated Sorted Array II
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
  if (nums[0] < nums[nums.length-1]) return nums[0];
  for (const val of nums) {
    if (val < nums[0]) return val;
  }
  return nums[0];
};

// Accepted
// 192/192 cases passed (99 ms)
// Your runtime beats 34.68 % of javascript submissions
// Your memory usage beats 17.74 % of javascript submissions (40.3 MB)
// @lc code=end

